### PR TITLE
Fix progress dialog's bars overflowing.

### LINKED ISSUE #10390: Several visual issues
**Operating system or device - Godot version:**

**Issue description:**

There are several visual issues
screenshot from 2017-08-16 23-09-03
screenshot from 2017-08-16 23-08-39
screenshot from 2017-08-16 23-08-29
screenshot from 2017-08-16 23-06-23

Ping @toger5 

**Steps to reproduce:**

**Link to minimal example project:**

### PR DESCRIPTION
**Pull request by groud**
Partly fixes #10390

### CODE REVIEW

File: editor/progress_dialog.cpp [hunk 1]
```diff
@@ -155,9 +155,10 @@ void ProgressDialog::_popup() {
 
 	Ref<StyleBox> style = get_stylebox("panel", "PopupMenu");
 	ms += style->get_minimum_size();
-	for (int i = 0; i < 4; i++) {
-		main->set_margin(Margin(i), style->get_margin(Margin(i)));
-	}
+	main->set_margin(MARGIN_LEFT, style->get_margin(MARGIN_LEFT));
+	main->set_margin(MARGIN_RIGHT, -style->get_margin(MARGIN_RIGHT));
```
Comment (akien-mga) [inline editor/progress_dialog.cpp]: Won't there be many other occurrences in the code where the margin is expected to be an absolute value?

Isn't it a bit unintuitive to expect users to define a negative right and bottom margin?

Comment (groud) [inline editor/progress_dialog.cpp]: Well, that's how the new anchoring system works. Now the margin is defined as the signed distance between the node side and the anchors. So here, as the right anchor is set at the end, you need a negative value for the right side of the node to be at the left of its anchor. Now controls' margins act more like "offsets" I'd say. (See our discussion with @toger5 in #9889)

Actually this makes the whole anchoring system more consistent, but this is more obvious when setting those values from the editor.

The fact is now that styleboxes and control nodes haven't the same conventions for margins anymore, while they are called the same. Maybe "margin" should be renamed to "offset".

**Comment by bojidar-bg** [comment]
BTW, next time, you should write `Fixes part of #...`, so that it won't autoclose it 😃

**Comment by groud** [comment]
@bojidar-bg Did not even know that this feature existed ^^

**Pull request merged**